processor/geoipprocessor/internal/provider/maxmindprovider/provider.go

```diff
@@ -39,7 +40,13 @@ func newMaxMindProvider(cfg *Config) (*maxMindProvider, error) {
 		return nil, fmt.Errorf("could not open geoip database: %w", err)
 	}
 
-	return &maxMindProvider{geoReader: geoReader, langCode: defaultLanguageCode}, nil
+	provider := &maxMindProvider{geoReader: geoReader, langCode: defaultLanguageCode}
+
+	runtime.AddCleanup(provider, func(geoReader *geoip2.Reader) {
+		_ = geoReader.Close()
+	}, provider.geoReader)
```

Comment:
I would prefer not use leverage `runtime` package to correctly handle cleanup which should be done in a shutdown call.

Reply:
Given components must have a `Stop` method makes sense to do the Close explicitly. I am about the change the PR, just testing out the new `runtime` method :)